source/material_model/visco_plastic.cc

```diff
@@ -180,14 +195,32 @@ namespace aspect
           // to compute the elastic force term.
           bool plastic_yielding = false;
           IsostrainViscosities isostrain_viscosities;
+
           if (in.requests_property(MaterialProperties::viscosity) || in.requests_property(MaterialProperties::additional_outputs))
             {
               // Currently, the viscosities for each of the compositional fields are calculated assuming
               // isostrain amongst all compositions, allowing calculation of the viscosity ratio.
               // TODO: This is only consistent with viscosity averaging if the arithmetic averaging
               // scheme is chosen. It would be useful to have a function to calculate isostress viscosities.
-              isostrain_viscosities =
-                rheology->calculate_isostrain_viscosities(in, i, volume_fractions, phase_function_values, n_phase_transitions_for_each_chemical_composition);
+              // Methods for averaging among phases depend on whether the most dominant phases are looked up
+              // for each composition in its respective lookup table. A separate phase function class
+              // manages the averaging if the user chooses this option.
+              if (use_dominant_phase_for_viscosity)
+                {
+                  std::vector<double> phase_function_discrete_values(phase_function_discrete->n_phase_transitions(), 0.0);
```

Comment:
but you already create this variable in line 116 above. I think this line can be deleted.